internal/olm/operator/bundle/install.go

```diff
@@ -87,6 +109,71 @@ func (i *Install) setup(ctx context.Context) error {
 		return err
 	}
 
+	var declcfg *declarativeconfig.DeclarativeConfig
+
+	// get index image labels.
+	catalogLabels, err := registryutil.GetImageLabels(ctx, nil, i.IndexImageCatalogCreator.IndexImage, false)
+	if err != nil {
+		return fmt.Errorf("get index image labels: %v", err)
+	}
+
+	// set the field to true if FBC label is on the image or for a default index image.
+	if _, hasFBCLabel := catalogLabels[containertools.ConfigsLocationLabel]; hasFBCLabel || i.IndexImageCatalogCreator.IndexImage == registry.DefaultIndexImage {
+		i.IndexImageCatalogCreator.HasFBCLabel = true
+	}
+
+	// generate an fbc if an fbc specific label is found on the image or for a default index image.
+	if i.IndexImageCatalogCreator.HasFBCLabel {
+		// FBC variables
+		f := &FBCContext{
+			Package:     labels[registrybundle.PackageLabel],
+			ChannelName: strings.Split(labels[registrybundle.ChannelsLabel], ",")[0],
+			Refs:        []string{i.BundleImage},
+			ChannelEntry: declarativeconfig.ChannelEntry{
+				Name: csv.Name,
+			},
+		}
+		log.Infof("Creating a File-Based Catalog of the bundle %q", i.BundleImage)
+
+		// generate a file-based catalog representation of the bundle image
+		declcfg, err = f.createFBC(ctx)
+		if err != nil {
+			log.Errorf("Error in generating file-based catalog with image %s: %v", i.BundleImage, err)
```

Comment:
If these errors occur, do we end up logging them twice? I wonder if we can return an error with the extra context you've included in the log message. Something like:
```go
return fmt.Errorf("error generating file-based catalog with image %q: %v", i.BundleImage, err)
```